threatmodels/notation-theatmodel.md

```diff
@@ -0,0 +1,85 @@
+  - Notation trust store, where certificates are added to validate signatures
+  - Notation plugin store, where Notation plugins are installed
+  - Credential store, where the registry or Key management credentials are stored.
+- **Notation Plugin**: An external component built by different vendors, and integrated to Notation as a signing plugin for signature generation, or as a verifying plugin for signature verification, see [plugin spec](https://github.com/notaryproject/notaryproject/blob/v1.0.0-rc.2/specs/plugin-extensibility.md) for details.
+- **Registry**: A remote registry of Docker and Open Container Initiative (OCI) images, with support for all OCI artifacts. It is the remote storage where artifacts are pushed to by artifact builder.
+- **KMS**: A remote Key Management Service provides key and associated certificate management, and remote signing service.
+
+## Notation sign artifacts using remote key
+
+### Data flow
+
+A signer uses Notation to sign artifacts with the private key stored in a KMS, and store signatures in a local file system or in a remote registry.
+
+The following diagram illustrates Notation signing remote artifacts with KMS:
+
+![image](../media/notation-sign-remote.png)
+
+The following diagram illustrates Notation signing local artifacts with KMS:
+
+![image](../media/notation-sign-local.png)
+
+According to [Notary v2 signature specification](../specs/signature-specification.md), the signing payload is a JSON document which requires the [OCI descriptor](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/descriptor.md) of the artifact. For artifacts stored in a registry, the descriptor info can be retrieved by inspecting the [image manifest](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/manifest.md) of the artifact. For artifacts stored locally as an [OCI Image layout](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/image-layout.md), the descriptor info can be retrieved from the `index.json` file from OCI Image layout. Users can also pass descriptor info as a JSON file to Notation directly.
+
+### Threats and Mitigation
+
+|   Title                      | STRIDE threat type     | Threat Status | Priority | Entity          | Description   | Mitigation   |
+| ---------------------------- | ---------------------- | ------------- | -------- | --------------- | ------------- | ------------ |
+| Compromised signing key      | Information disclosure | Mitigated     | High     | Signer          | Signing keys are compromised, and this may lead to arbitrary artifacts being signed by attackers | Rotate the signing key, and revoke the associated certificates |
+| Inaccessible Registry        | Denial of Service      | Mitigated     | High     | Registry        | Registry is not able to service incoming requests or perform up to spec, thus users are unable to publish any artifacts | Users sign offline artifacts and publish via other medias |
```

Comment:
Can we revise the Mitigation "Users sign offline artifacts and publish via other medias" to "Users sign local artifacts offline and publish via other medias", which aligns with the term "local artifacts" we used above?